opennms-provision/opennms-detector-common/src/test/java/org/opennms/netmgt/provision/detector/common/DetectorRequestDTOTest.java

```diff
@@ -0,0 +1,43 @@
+package org.opennms.netmgt.provision.detector.common;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.junit.runners.Parameterized.Parameters;
+import org.opennms.core.test.xml.XmlTestNoCastor;
+
+
+public class DetectorRequestDTOTest  extends XmlTestNoCastor<DetectorRequestDTO>{
+
+    public DetectorRequestDTOTest(DetectorRequestDTO sampleObject,
+            String sampleXml) {
+        super(sampleObject, sampleXml, null);
+    } 
+    
+    @Parameters
+    public static Collection<Object[]> data() throws Exception {
+        return Arrays.asList(new Object[][] {
+                {
+                    getDetectorRequest(),
+                    "<?xml version=\"1.0\"?>\n" +
+                    "<detector-request location=\"MINION\" serviceName=\"ICMP\" address=\"localhost\">\n"+
+                    "<attributes>port=8980</attributes> \n"+
```

Comment:
We should encode the attributes as key value pairs instead of concatenated strings i.e.
```
<attribute name="port">8980</attribute>
```